pkg/collector/corechecks/oracle-dbm/init.go

```diff
@@ -62,18 +62,16 @@ func (c *Check) init() error {
 
 	if c.config.ReportedHostname != "" {
 		c.dbHostname = c.config.ReportedHostname
-	} else {
-		if i.HostName.Valid {
-			c.dbHostname = i.HostName.String
-		} else {
-			// host_name is null on Oracle Autonomous Database
-			c.dbHostname = i.InstanceName
-		}
+	} else if i.HostName.Valid {
+		c.dbHostname = i.HostName.String
 	}
 	if i.HostName.Valid {
 		tags = append(tags, fmt.Sprintf("real_hostname:%s", i.HostName.String))
 	}
-	tags = append(tags, fmt.Sprintf("host:%s", c.dbHostname), fmt.Sprintf("oracle_version:%s", c.dbVersion))
+	if c.dbHostname != "" {
+		tags = append(tags, fmt.Sprintf("host:%s", c.dbHostname), fmt.Sprintf("db_server:%s", c.dbHostname))
+	}
+	tags = append(tags, fmt.Sprintf("database_instance:%s/%s", c.dbHostname, c.cdbName), fmt.Sprintf("oracle_version:%s", c.dbVersion))
```

Comment:
🤔 That seems wrong to me. The `database_instance` tag is usually not one that we set directly on metrics. It's inferred by metrics-intake from _resources_ that are attached to metrics. Doing it directly here would bypass the resource aliasing/tag resolution for the `database_instance` resource but it would also be confusing because we might reasonably think that the `database_instance` tag means the resource was sent on metrics.

Reply:
@nenadnoveljic sorry i was not thinking this morning when i approved this PR, but this is not the right approach. The `database_instance` stuff is a tag that is created and applied in our backend via the creation of metadata resources. You should not be adding this as a tag, but rather sending a `database_instance` for this host, similarirly to what we do in the postgres integration here:

https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/postgres.py#L829-L848

Reply:
OK, sorry, I misunderstood the requirement defined [here](https://dd.slack.com/archives/C04HV5BPBU2/p1696348193127319?thread_ts=1695127160.589659&cid=C04HV5BPBU2).

Reply:
Sorry that requirement wasn't clear. The `database_instance` _tag_ is added by metrics-intake based on resources. Setting the tag along without adding the resource would result in confusion because we might initially think that this would be from metrics-intake from a resource when it would just have been directly emitted by the oracle integration. 

[This diagram](https://docs.google.com/document/d/1onXhg-XVob38C1nHurzLC-s_QqKcmXW4QQdFbMkPuVM/edit#heading=h.b9dubcq0z8jp) might help clear things up a bit but you'll still have to refer to the postgres implementation as an example because the internal tags are used as a mechanism to carry resources in the agent internals and up to the point where the metrics payload for metrics-intake is generated. 

> But to what value should I set host on the agent? In Postgres, host is set to the servername, but in Oracle, we want to charge per db, so it should be set to servername/cdb in the end. Should this be done on the backend or on the agent?

Postgres is not the best example for this special case (even though it is a good example for how to set `database_instance` as a _resource_). But sqlserver and the azure sqlserver database is a more similar example. For that one, [we do this in the integration](https://github.com/DataDog/integrations-core/blob/master/sqlserver/datadog_checks/sqlserver/sqlserver.py#L296-L299). I think this is the best way of doing this as it covers _all_ metrics and not just the metrics emitted by the dbm backend. But we can also do some normalization in the backend for older agent version and set the hostname _and_ database_instance to `servername/cdb`.

Reply:
Thanks! 

So the agent basically needs to send the tag `dd.internal.resource:database_instance:<HOST>/<CDB>` with every metric and event?  

Do we also need to send this [metadata event](https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/postgres.py#L862-L881)?

Reply:
Nice, thanks for doing the full thing, @nenadnoveljic ! I checked out metrics and confirmed we've seen oracle database_instance payloads processed earlier today.